cmd/isolateDAG.go

```diff
@@ -0,0 +1,174 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"ddbt/config"
+	"ddbt/fs"
+	"ddbt/utils"
+)
+
+func init() {
+	rootCmd.AddCommand(isolateDAG)
+	addModelsFlag(isolateDAG)
+}
+
+var isolateDAG = &cobra.Command{
+	Use:   "isolate-dag",
+	Short: "Creates a symlinked copy of the selected models, which can be then passed to Fishtown's DBT",
+	Run: func(cmd *cobra.Command, args []string) {
+		fileSystem, _ := compileAllModels()
+
+		graph := buildGraph(fileSystem, ModelFilter) // Build the execution graph for the given command
+		graph.AddReferencingTests()                  // And then add any tests which reference that graph
+
+		if err := graph.AddAllUsedMacros(); err != nil {
+			fmt.Printf("❌ Unable to get all used macros: %s\n", err)
+			os.Exit(1)
+		}
+
+		isolateGraph(graph)
+	},
+}
+
+func isolateGraph(graph *fs.Graph) {
+	pb := utils.NewProgressBar("🔪 Isolating DAG", graph.Len())
+	defer pb.Stop()
+
+	// Create a temporary directory to stick the isolated models in
+	isolationDir, err := ioutil.TempDir(os.TempDir(), "isolated-dag-")
+	if err != nil {
+		fmt.Printf("❌ Unable to create temporarily directory for DAG isolation: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Get the current working directory
+	cwd, err := os.Getwd()
+	if err != nil {
+		fmt.Printf("❌ Unable to get working directory: %s\n", err)
+		os.Exit(1)
+	}
+
+	symLink := func(pathInProject string) error {
+		fullOrgPath := filepath.Join(cwd, pathInProject)
+		symlinkedPath := filepath.Join(isolationDir, pathInProject)
+
+		// Create the folder in the isolated dir if needed
+		err := os.MkdirAll(filepath.Dir(symlinkedPath), os.ModePerm)
+		if err != nil {
+			return err
+		}
+
+		// Symlink the file in there
+		err = os.Symlink(fullOrgPath, symlinkedPath)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	// Create a blank file which DBT can read
+	touch := func(pathInProject string) error {
```

Comment:
I'm in two minds about whether this and `symlink` should be defined inline like this or if they should be function declarations but happy with them as they are 👍

Reply:
I defined them inline as I wanted a cleaner call without having to pass the current working directory or isolation directory variables in each time I called them